### PR TITLE
Sendwp

### PR DESCRIPTION
**Pull request by kjohnson**
Add SendWP recommendations.

## TODO
- [x] Update sales copy for Email Setting banner ad.
- [x] Update outbound links with UTM Tracking.

## Email Settings

- Adds a "notification nudge" on Email Settings (on a 2 second delay).

image

- Replaces the deprecated notice with a banner ad.

image

## Email Processor

- Updates email service recommendation to SendWP.
- Adds a banner ad for SendWP.

image

### CODE REVIEW

File: sendwp/handler.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
+<?php
+
+add_action( 'wp_ajax_caldera_forms_sendwp_remote_install', 'wp_ajax_caldera_forms_sendwp_remote_install_handler' );
```
Comment (New0) [inline sendwp/handler.php]: Could you comment the function or purpose of the action ?

File: sendwp/installer.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+function caldera_forms_sendwp_remote_install() {
```
Comment (New0) [inline sendwp/installer.js]: I think it would also help to have the three functions of the file commented

File: sendwp/init.php [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+include_once CFCORE_PATH . 'sendwp/handler.php';
+
+function caldera_forms_admin_enqueue_sendwp_installer() {
```
Comment (New0) [inline sendwp/init.php]: Could the the function be commented too ?

File: sendwp/handler.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        echo json_encode( array(
+            'error' => __( 'Something went wrong. SendWP was not installed correctly.' ),
+            'install' => $install,
+            ) );
+        exit;
+    }
+    
+    echo json_encode( array(
+        'partner_id' => 2400,
+        'register_url' => sendwp_get_server_url() . '_/signup',
```
Comment (New0) [inline sendwp/handler.php]: We may want to add UTM data to the url there ?

File: sendwp/init.php (hunk 3 above)
Comment (kjohnson) [reply sendwp/init.php]: Added a dockblock comment.

File: sendwp/handler.php (hunk 1 above)
Comment (kjohnson) [reply sendwp/handler.php]: I added a README file for the SendWP SDK usage.

**Comment by kjohnson** [comment]
@New0 I updated the base branch to `develop`, not seeing any merge conflicts 😅 

Also, sorry again for not explaining the SendWP SDK setup. I should have created an issue with that explanation, which would have greatly benefitted the review process.

File: sendwp/handler.php [hunk 5]
```diff
@@ -0,0 +1,73 @@
+<?php
+
+add_action( 'wp_ajax_caldera_forms_sendwp_remote_install', 'wp_ajax_caldera_forms_sendwp_remote_install_handler' );
+function wp_ajax_caldera_forms_sendwp_remote_install_handler () {
+    $all_plugins = get_plugins();
+    $is_sendwp_installed = false;
```
Comment (Shelob9) [inline sendwp/handler.php]: SECURITY There is no nonce check or authorization check. Please using existing APIs for capability checks and nonce generation.

File: sendwp/handler.php [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        include_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+        
+        /*
+        * Use the WordPress Plugins API to get the plugin download link.
+        */
+        $api = plugins_api( 'plugin_information', array(
+            'slug' => $plugin_slug,
+        ) );
+        if ( is_wp_error( $api ) ) {
+            ob_end_clean();
```
Comment (Shelob9) [inline sendwp/handler.php]: SECURITY must escape all strings returned to DOM

File: sendwp/handler.php [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+            ) );
+        exit;
+    }
+    
+    echo json_encode( array(
+        'partner_id' => 2400,
+        'register_url' => sendwp_get_server_url() . '_/signup',
+        'client_name' => sendwp_get_client_name(),
+        'client_secret' => sendwp_get_client_secret(),
+        'client_redirect' => sendwp_get_client_redirect(),
```
Comment (Shelob9) [inline sendwp/handler.php]: SECURITY not escaping URLs is a risk for cross site scripting attacks.

File: ui/panels/emailer.php [hunk 8]
```diff
@@ -197,8 +197,17 @@
...
+					<?php echo sprintf( esc_html( 'If you are having email issues, we strongly recommend %sSendWP%s.', 'caldera-forms' ), '<a href="https://sendwp.com?utm_source=Caldera+Forms+Plugin&utm_medium=Forms_Edit+Forms_Email&utm_campaign=SendWP+banner+ad" target="_blank" rel="nofollow">', '</a>' ); ?>
 				</p>
+
+				<a href="https://sendwp.com?utm_source=Caldera+Forms+Plugin&utm_medium=Forms_Edit+Forms_Email&utm_campaign=SendWP+banner+ad" target="_blank" rel="nofollow" style="text-decoration:none;">
+					<div class="mailer_config_panel caldera-config-processor-notice" style="clear: both; padding: 20px 0px 0px;width:550px;">
+						<p style="padding:12px;text-align:center;color:white;background:#21394a;" class="description">
+							<?php echo sprintf(__('%sSendWP%s - Fix Your WordPress Email%sThe easy solution to transactional email in WordPress', 'caldera-forms'), '<strong>', '</strong>', '<br />' ); ?>
```
Comment (Shelob9) [inline ui/panels/emailer.php]: SECURITY unescaped translation string

**Pull request merged**